src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,17 @@
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidDataException("invalid data");
+        } else if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new InvalidDataException("problem with login");
+        } else if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new InvalidDataException("problem with password");
+        } else if (user.getAge() < 18) {
+            throw new InvalidDataException("problem with age");
+        } else if (storageDao.get(user.getLogin()) == null) {
+            return storageDao.add(user);
```

Comment:
The `add` method should be called only if the user does not already exist. Ensure that the logic correctly reflects this by adjusting the condition on line 20.